pshchegolevatykh/LinkedListCycle.cs

```diff
@@ -0,0 +1,92 @@
+namespace ConsoleApp
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            var givenList = new ListNode(5);
+            InsertToTail(givenList, 6);
+            InsertToTail(givenList, 7);
+            InsertToHead(ref givenList, 8);
+            InsertToHead(ref givenList, 9);
+            InsertAfter(givenList.next, 11);
+            DeleteFirst(ref givenList);
+
+            bool hasCycle = HasCycle(givenList);
+            Console.WriteLine($"Given list has cycle? {hasCycle}");
+            Console.WriteLine("Okay let's make one.");
+            givenList.next.next.next = givenList.next;
+            hasCycle = HasCycle(givenList);
+            Console.WriteLine($"Given list has cycle? {hasCycle}");
+        }
+
+        public static void InsertToTail(ListNode head, int itemToInsert)
+        {
+            while (head.next != null)
+            {
+                head = head.next;
+            }
+
+            head.next = new ListNode(itemToInsert);
+        }
+
+        public static void InsertAfter(ListNode givenNode, int itemToInsert)
+        {
+            if (givenNode == null)
+            {
+                return;
+            }
+
+            ListNode newNode = new ListNode(itemToInsert);
+            newNode.next = givenNode.next;
+            givenNode.next = newNode;
+        }
+
+        public static void DeleteFirst(ref ListNode head)
+        {
+            if (head != null)
+            {
+                head = head.next;
+            }
+        }
+
+        public static void InsertToHead(ref ListNode head, int itemToInsert)
+        {
+            var newHead = new ListNode(itemToInsert);
+            newHead.next = head;
+            head = newHead;
+        }
+
+        public static bool HasCycle(ListNode head)
+        {
+            var visitedNodes = new List<ListNode>();
+
+            while(head != null)
+            {
+                if (visitedNodes.Contains(head))

```

Comment:
who uses list structure for search 
![image](https://user-images.githubusercontent.com/35116217/179709313-d8ff4a53-e1c7-46a8-9704-159f9c702177.png)

Reply:
although, memory usage is good somehow 🤔

Reply:
@quadreff typical hash set fan and list average enjoyer

![image](https://user-images.githubusercontent.com/48563768/179714393-6b9a1aa5-8821-40fb-9934-fc70dbd47c59.png)

Reply:
https://www.youtube.com/watch?v=Ux5cQbO_ybw

Reply:
Yea somehow I managed to find data structure where insertion is the bottleneck. But leetcode said it's okay so I blame them for not checking this. So far it's the most NAIVE solution. Keep those memes coming!

Reply:
Actually Add() is O(1) for List<T> though Contains is definitely O(N) 🙂